[noahxinjie] iP

noahxinjie opened the pull request:
# Jizo


> "Den bor blant oss!" - sussy baka

Jizo is a friendly bot that frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~fast~~ _SUPER_ FAST to use brrr

All you need to do is,

- download it from [here](https://github.com/noahxinjie/ip)
- double click it
- tell it to add your tasks
- ???
- **PROFIT!** 😉

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon!!)
- [ ] Reminders (coming soon!!)

***
You can go to the `main` method in `Launcher.java` to run Jizo manually:
```
public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
```

Commit:
Add Gradle support

noahxinjie committed:
Level-1

noahxinjie committed:
Level-2

noahxinjie committed:
Level-3

noahxinjie committed:
Level-4

noahxinjie committed:
A-TextUiTesting and Edit existing issue with list cmd creating newline

noahxinjie committed:
Level-5

noahxinjie committed:
Level-6

noahxinjie committed:
Make styling and other minor edits

noahxinjie committed:
Add ability to save tasks

noahxinjie committed:
Modify storage of deadlines dates

noahxinjie committed:
Allow program to catch DateTimeParseException when parsing date inputs

noahxinjie committed:
Merge branch 'branch-Level-7'

noahxinjie committed:
Merge branch 'branch-Level-8'

noahxinjie committed:
Add more OOP

noahxinjie committed:
Organise classes into packages

noahxinjie committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

noahxinjie committed:
Add Junit tests for Task classes

noahxinjie committed:
Remove unnecessary dependency in build.gradle

noahxinjie committed:
Modify build.gradle to create JAR file with Gradle

noahxinjie committed:
Add Javadocs for most classes and methods

noahxinjie committed:
Modify code to comply with coding standards

noahxinjie committed:
Add ability to find tasks

noahxinjie committed:
Merge branch 'branch-A-JavaDoc'

noahxinjie committed:
Add data/ to .gitignore

noahxinjie committed:
Merge branch 'branch-Level-9'

noahxinjie committed:
Improve JavaDocs

dfordarius commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for an instance of Duke.
+     * Contains a Ui for interacting with user,
+     * Storage for saving and retrieving tasks from a file,
+     * and Tasklist for storing the tasks.
+     * Catches any errors thrown when the program is running.
+     * @param filePath
```
Should there be a new line between your description and @param line? 
Should there be no punctuation after each parameter description?

dfordarius commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+            String taskType = inputs[0];
+            String status = inputs[1];
+            boolean isDone = status.equals("[X");
+            String rest = inputs[2];
+
+            if (taskType.equals("[T")) {
```
Might be an accidental extra empty line here

dfordarius commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,38 @@
+ * Represents the command to add Todo, Deadline and Event tasks.
+ */
+public class AddCommand extends Command {
+    private Task t;
+
```
Perhaps you can name "Task t" to something like "Task task" as variables with large scope should have long names

dfordarius reviewed (commented):
In general, I think most of the coding standard requirements are adhered to! I noticed that you have not completed the A-Checkstyle increment yet, so your styling has not adjusted to the checkstyle so its understandable that your styling differs abit from the checkstyle standard. Keep up the good work, jiayou!

noahxinjie committed:
Implement GUI

noahxinjie committed:
Change images

noahxinjie committed:
Merge branch 'branch-Level-10'

noahxinjie committed:
Configure CheckStyle

noahxinjie committed:
Improve code to fit with CheckStyle standards

noahxinjie committed:
Merge branch 'branch-A-CheckStyle'

noahxinjie committed:
Update JavaDocs

WesleyBLDC commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,53 @@
+        this.tasks = new ArrayList<>();
+    }
+
+    public void add(Task t) {
```
Suggestion: change t to task so that is clear to read as the code gets longer or gets used in other places

WesleyBLDC commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+    /**
+     * Creates a storage file at the intended file path.
+     */
+    public Storage() {
```
Is this redundant?

WesleyBLDC replied:
Or can add an extra constructor for when user wants to allocate multiple chiwas with each having their own storage.

WesleyBLDC commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+        ArrayList<Task> tasks = new ArrayList<>();
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] inputs = line.split("]", 3);
```
Suggestion: Perhaps would be good to check the char at index 2 instead? so there might be less manipulation of the [].

WesleyBLDC reviewed (approved):
Overall the code looks good, and also included javadocs with proper formatting

LGTM

noahxinjie committed:
Edit build.gradle to build jar files

noahxinjie committed:
Add Assertions

There are currently no assert statements to double check for
program's correctness.

Adding assert statements helps to document important assumptions that
should hold at various points in the code.

Let's add assert statements to Storage, Parser and TaskList classes.

In the future, more assert statements can be included to
keep up with code updates.

noahxinjie committed:
Refactor code for better code quality

Currently, there are a few methods that are lengthy and
contain deep nesting. This has negatively impacted
code readability.

Refactoring code will help improve code quality.

Let's,
- Extract methods out from long methods
- Replace Nested Conditional with guard clauses to reduce
unnecessary indentation

noahxinjie committed:
Add JavaDocs for newly extracted methods

noahxinjie committed:
Merge pull request #1 from noahxinjie/branch-A-Assertions

Add Assertions

noahxinjie committed:
Merge branch 'master' into branch-A-CodeQuality

noahxinjie committed:
Merge pull request #2 from noahxinjie/branch-A-CodeQuality

Refactor code for better code quality

noahxinjie committed:
Set up CI with GitHub Actions

ongweijie7 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+                String byString = addon[1];
+                String[] byPart = byString.split(" ", 2);
+                String by = byPart[1];
+                task = new Deadline(description, by);
```
Maybe there can be better naming convention to avoid any confusion and to allow you to also understand/recall the variables better

ongweijie7 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,17 @@
+package duke.task;
+
+public class Event extends Task {
+    protected String from;
+    protected String to;
```
Maybe this could be named as 
```suggestion
    protected String fromDate;
    protected String toDate;
```
Might this be better in helping you remember the variables?

ongweijie7 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+                String toString = addon[2];
+                String[] toPart = toString.split(" ", 2);
+                String to = toPart[1];
+                task = new Event(description, from, to);
```
There's some arrowheaded style code here. You may want to consider seperating the conditions instead of lumping both of them under else

ongweijie7 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+    /**
+     * Saves user-inputted tasks into a file.
+     *
+     * @param tasks to be saved
+     * @throws IOException if file cannot be accessed
```
Don't forget your periods at the end of each sentence for java docs

ongweijie7 reviewed (commented):
In general, I think it was a pretty good job at the following the naming conventions and coding standards taught. Besides that, do remember to leave a period at the end of your javadocs comments and avoiding deep nesting although yours was quite trivial. Good job 👍 👍

noahxinjie committed:
Add capability to indicate priority for tasks

noahxinjie committed:
Add new reply to user when list is empty

noahxinjie committed:
Improve GUI

noahxinjie committed:
Add app icon

noahxinjie committed:
Merge branch 'branch-A-BetterGui'

noahxinjie committed:
Modify app name and replies to user

noahxinjie committed:
Allow parser to handle more user input errors

noahxinjie committed:
Resolve issue with truncated long texts in DialogBox

noahxinjie committed:
Resolve issue with incorrect parsing of tasks' priority descriptions

noahxinjie committed:
Add product screenshot and user guide

noahxinjie committed:
Set up jekyll theme

noahxinjie committed:
Change title of GitHub page

noahxinjie committed:
Change name of JAR file

noahxinjie committed:
Resolve issue with not being able to access data folder

noahxinjie committed:
Resolve CheckStyle errors

noahxinjie committed:
Update JUnit tests

noahxinjie committed:
Remove I/O redirection tests from GitHub Actions

noahxinjie committed:
Remove redundant Storage constructor
